sound/soc/codecs/rt5682.c

```diff
@@ -879,8 +911,14 @@ static void rt5682_enable_push_button_irq(struct snd_soc_component *component,
 		snd_soc_component_update_bits(component, RT5682_4BTN_IL_CMD_2,
 			RT5682_4BTN_IL_MASK | RT5682_4BTN_IL_RST_MASK,
 			RT5682_4BTN_IL_EN | RT5682_4BTN_IL_NOR);
+#if !IS_ENABLED(CONFIG_SOUNDWIRE)
 		snd_soc_component_update_bits(component, RT5682_IRQ_CTRL_3,
 			RT5682_IL_IRQ_MASK, RT5682_IL_IRQ_EN);
+#else
+		snd_soc_component_update_bits(component, RT5682_IRQ_CTRL_3,
+			RT5682_IL_IRQ_MASK | RT5682_IL_IRQ_TYPE_MASK,
+			RT5682_IL_IRQ_EN | RT5682_IL_IRQ_PUL);
+#endif
```

Comment:
I think you need a private field/variable that's set when you probe in SoundWire mode, that would help you select the right path